sdl2/chase_in_space/chase_in_space.odin

```diff
@@ -0,0 +1,229 @@
+/*
+	Simple game about enemies chasing and shooting you in 2D space.
+
+	Original repo: https://github.com/wbogocki/Odin-Play
+
+	To play:
+		- Move with WSAD or arrow keys
+		- Jump with Shift + direction
+*/
+
+package chase_in_space
+
+import "core:fmt"
+import "core:math"
+import "vendor:sdl2"
+
+Game :: struct {
+	renderer: ^sdl2.Renderer,
+	keyboard: []b8,
+	time:     f64,
+	dt:       f64,
+	entities: [dynamic]Entity,
+}
+
+EntityType :: enum {
+	PLAYER,
+	ENEMY,
+	PROJECTILE,
+}
+
+Entity :: struct {
+	type:           EntityType,
+	hp:             int,
+	pos:            [2]f32,
+	vel:            [2]f32,
+	reload_counter: f32,
+	bullet_decay:   f32,
+	dash_counter:   f32,
+}
+
+render_entity :: proc(entity: ^Entity, game: ^Game) {
+	switch entity.type {
+	case .PLAYER:
+		sdl2.SetRenderDrawColor(game.renderer, 255, 0, 255, 0)
+		sdl2.RenderDrawRectF(
+			game.renderer,
+			&sdl2.FRect{x = entity.pos.x, y = entity.pos.y, w = 10, h = 10},
+		)
+	case .ENEMY:
+		sdl2.SetRenderDrawColor(game.renderer, 0, 255, 255, 0)
+		sdl2.RenderDrawRectF(
+			game.renderer,
+			&sdl2.FRect{x = entity.pos.x, y = entity.pos.y, w = 10, h = 10},
+		)
+	case .PROJECTILE:
+		sdl2.SetRenderDrawColor(game.renderer, 0, 255, 255, 0)
+		sdl2.RenderDrawPointF(game.renderer, entity.pos.x, entity.pos.y)
+	}
+}
+
+find_entity :: proc(type: EntityType, game: ^Game) -> ^Entity {
+	for _, i in game.entities {
+		if game.entities[i].type == type {
+			return &game.entities[i]
+		}
+	}
+	return nil
+}
+
+update_entity :: proc(entity: ^Entity, game: ^Game) {
+	dt := f32(game.dt)
+	switch entity.type {
+	case .PLAYER:
+		dir := [2]f32{0, 0}
+		if game.keyboard[sdl2.SCANCODE_UP] | game.keyboard[sdl2.SCANCODE_W] do dir.y -= 1
+		if game.keyboard[sdl2.SCANCODE_DOWN] | game.keyboard[sdl2.SCANCODE_S] do dir.y += 1
+		if game.keyboard[sdl2.SCANCODE_LEFT] | game.keyboard[sdl2.SCANCODE_A] do dir.x -= 1
+		if game.keyboard[sdl2.SCANCODE_RIGHT] | game.keyboard[sdl2.SCANCODE_D] do dir.x += 1
```

Comment:
I'd probably do some basic alignment here to make it easier to read.

Reply:
This and the one at the bottom were auto-formatted with odin-fmt, should I still align them?